extensions/amp-analytics/0.1/test/test-video-analytics.js

```diff
@@ -0,0 +1,573 @@
+        undefined,
+        AnalyticsEventType.VIDEO, {
+          'on': 'video-percentage-played',
+          'selector': selectors,
+          'videoSpec': {
+            'percentages': [5, 25, 50, 75, 100]
+          }
+        },
+        fn1
+      );
+
+      let data = {
+        'autoplay': false,
+        'currentTime': 0.0045,
+        'duration': 15,
+        'height': 399,
+        'id': 'myVideo',
+        'muted': false,
+        'normalizedPercentage': "100",
+        'playedRangesJson': "[[0,0.0045]]",
+        'playedTotal': 0.0045,
+        'state': 'paused',
+        'width': 700,
+      };
+
+      let data_2 = {
+        'autoplay': false,
+        'currentTime': 0.0045,
+        'duration': 15,
+        'height': 399,
+        'id': 'myVideo-2',
+        'muted': false,
+        'normalizedPercentage': "50",
+        'playedTotal': 0.0045,
+        'playedRangesJson': "[[0,0.0045]]",
+        'state': 'playing_manual',
+        'width': 700,
+      };
+
+      dispatchCustomEvent(myVideo, 'video-percentage-played', data, null)
+      dispatchCustomEvent(myVideo2, 'video-percentage-played', data_2, null)
+
+      await macroTask();
+      expect(fn1).to.have.callCount(2);
+      expect(fn1.firstCall).to.be.calledWith(new AnalyticsEvent([myVideo], 'video-percentage-played', data))
+      expect(fn1.secondCall).to.be.calledWith(new AnalyticsEvent([myVideo2], 'video-percentage-played', data_2))
+      expect(getElementSpy).to.be.callCount(2);
+      expect(tracker.sessionObservable_.handlers_.length).to.equal(1);
+
+    });
+
+    it('fires on video-seconds-played trigger', async () => {
+      const fn1 = env.sandbox.stub();
+      const fn2 = env.sandbox.stub();
+      let getElementSpy = env.sandbox.spy(root, 'getElement');
+
+      tracker.add(
+        undefined,
+        AnalyticsEventType.VIDEO, {
+          'on': 'video-seconds-played',
+          'selector': selectors,
+          'videoSpec': {
+            "end-session-when-invisible": false,
+            "exclude-autoplay": false,
+            'interval': 7
+          }
+        },
+        fn1
+      );
+
+      let data = {
+        'autoplay': false,
+        'currentTime': 10,
+        'duration': 15,
+        'height': 399,
+        'id': 'myVideo',
+        'muted': false,
+        'playedRangesJson': "[[0,10.498756]]",
+        'playedTotal': 10.498756,
+        'state': 'playing_manual',
+        'width': 700,
+      };
+
+      let data_2 = {
+        'autoplay': false,
+        'currentTime': 7.49,
+        'duration': 15,
+        'height': 399,
+        'id': 'myVideo-2',
+        'muted': false,
+        'playedTotal': 0.0045,
+        'playedRangesJson': "[[0,7.49]]",
+        'state': 'playing_manual',
+        'width': 700,
+      };
+
+      dispatchCustomEvent(myVideo, 'video-seconds-played', data, null)
+      dispatchCustomEvent(myVideo2, 'video-seconds-played', data_2, null)
+
+      await macroTask();
+      // expect(fn1).to.have.callCount(2);
```

Comment:
I am having trouble with the video-seconds-played trigger. For some reason, the commented out statements produce an error because the stub (fn1) has not been called at all. 
To find the error, I tried printing out the event at places it would be fired (observable.js and events.js) and also compared the event object with one produced by a real video element. The event gets printed out at places that it should be fired and also has a similar structure to a real video's event object. 
I would appreciate any insight for where to look.